### PR TITLE
Mobile

### PR DESCRIPTION
**Pull request by 4-life**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,3 +48,13 @@ For production build run:
     $ gulp build
 
 This will compile `assistant.user.js` file with minification and uglify.
+
+## Mobile build
+
+Mobile build is [here](https://AdguardTeam.github.io/AdguardAssistant/mobile/assistant.user.js)
+
+In the mobile assistant, the rules are specified via a function ADG_addRule:
```
Comment (ameshkov) [inline README.md]: Mobile build should not be a userscript, just a script that you can load & run.

Signature was described in the issue

File: compiler.meta.template.mobile.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// ==UserScript==
```
Comment (ameshkov) [inline compiler.meta.template.mobile.js]: No need

File: gulpfile.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ gulp.task('dev', () => {
 gulp.task('mobile', () => {
     options.debug = true;
     options.metaPath = options.metaDevMobile;
-    runSequence('css', 'compile-mobile', 'preprocess', 'restore-meta');
+    runSequence('css', 'compile-mobile', 'preprocess', 'uglify');
```
Comment (ameshkov) [inline gulpfile.js]: No need in minification

File: README.md [hunk 4]
```diff
@@ -48,3 +48,15 @@ For production build run:
     $ gulp build
 
 This will compile `assistant.user.js` file with minification and uglify.
+
+## Mobile build
+
+Mobile build is [here](https://AdguardTeam.github.io/AdguardAssistant/mobile/script.js)
```
Comment (ameshkov) [inline README.md]: Should be updated with the proper instruction

File: gulpfile.js [hunk 5]
```diff
@@ -33,6 +34,13 @@ gulp.task('dev', () => {
     runSequence('css', 'compile', 'preprocess', 'restore-meta');
 });
 
+gulp.task('mobile', () => {
+    options.debug = true;
+    options.metaPath = options.metaDevMobile;
+    options.scriptName = 'script';
+    runSequence('css', 'compile-mobile', 'preprocess', 'uglify');
```
Comment (ameshkov) [inline gulpfile.js]: No need in minification, really

File: locales/compiler.meta.dev.mobile.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
+// ==UserScript==
```
Comment (ameshkov) [inline locales/compiler.meta.dev.mobile.js]: Why do we need it?

File: src/controllers/SliderMenuControllerMobile.js [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        };
+        Object.keys(menuEvents).forEach(function (item) {
+            $(contentDocument.querySelectorAll(item)).on('click', menuEvents[item]);
+        });
+    };
+
+    var blockElement = function () {
+        addRule(getFilterRuleInputText());
+        iframeCtrl.removeIframe();
+        CommonUtils.reloadPageBypassCache();
```
Comment (ameshkov) [inline src/controllers/SliderMenuControllerMobile.js]: Page reloading is unnecessary, let's move it to the default `addRule` implementation.

File: locales/compiler.meta.dev.mobile.js (hunk 6 above)
Comment (4-life) [inline locales/compiler.meta.dev.mobile.js]: Just for @require string which includes the necessary scripts

File: src/controllers/SliderMenuControllerMobile.js [hunk 8]
```diff
@@ -89,7 +89,6 @@ var SliderMenuControllerMobile = function ($, selector, sliderWidget, settings,
     var blockElement = function () {
         addRule(getFilterRuleInputText());
         iframeCtrl.removeIframe();
-        CommonUtils.reloadPageBypassCache();
```
Comment (ameshkov) [inline src/controllers/SliderMenuControllerMobile.js]: That's for mobile only, right?

File: tasks/update-meta-locales.js [hunk 9]
```diff
@@ -13,6 +13,7 @@ module.exports = () => {
     const options = global.options || {};
 
     let compilerMeta = fs.readFileSync('compiler.meta.template.js').toString();
+    let compilerMetaMobile = fs.readFileSync('compiler.meta.template.js').toString();
```
Comment (atropnikov) [inline tasks/update-meta-locales.js]: `compiler.meta.template.mobile.js`?

File: locales/compiler.meta.dev.mobile.js (hunk 6 above)
Comment (ameshkov) [inline locales/compiler.meta.dev.mobile.js]: At least indicate it in a comment and remove all the unused directives

File: compiler.meta.template.mobile.js [hunk 10]
```diff
@@ -1,24 +1,8 @@
...
-// @exclude *://mil.ru/*
-// @exclude *://feedly.com/*
-// @exclude *wikipedia.org*
-// @exclude *icloud.com*
-// @exclude *hangouts.google.com*
-// @exclude *www.facebook.com/plugins/сomments*
-// @exclude *disqus.com/embed/comments*
-// @exclude *vk.com/widget_comments*
-// @exclude *www.youtube.com/embed/*
 // @require src\utils\css.escape.js
```
Comment (ameshkov) [inline compiler.meta.template.mobile.js]: Add a comment that explains why do we need this file

File: src/button.mobile.js [hunk 11]
```diff
@@ -28,8 +27,18 @@ var UIButtonMobile = function(log, settings, uiValidationUtils, $, gmApi, uiUtil
         }
         log.debug("Requirements checked, all ok");
         button = $(resources.getResource('button.html'));
-        gmApi.GM_addStyle(resources.getResource('button.css'));
-        gmApi.GM_addStyle(resources.getResource('selector.css'));
+
+        var css = document.createElement('style');
```
Comment (ameshkov) [inline src/button.mobile.js]: That's weird, why removing the `gmApi`? You could've simply supplied your own implementation in the mobile build.

File: src/main.mobile.js [hunk 12]
```diff
@@ -1,11 +1,16 @@
 /* global Ioc, Log, Resources, Settings, UIValidationUtils, balalaika, UIUtils, Localization, IframeControllerMobile, SliderWidget, AdguardRulesConstructorLib, AdguardSelectorLib, UIButtonMobile */
 
-var adguardAssistant = (function() {
+// var GM_getValue = function() {return false;};
```
Comment (ameshkov) [inline src/main.mobile.js]: ?

File: test/mobile.html [hunk 13]
```diff
@@ -23,13 +23,10 @@
...
         </script>
 
+        <script src="../build/script.js"></script>
+
         <script>
-            window.GM_getValue = function() {return false;};
-            window.GM_setValue = function() {return false;};
-            window.GM_getResourceText = function() {return false;};
-            window.GM_addStyle = function() {return false;};
+            adguardAssistant.start(console.log('check'));
```
Comment (ameshkov) [inline test/mobile.html]: What does this test check?

File: src/main.mobile.js (hunk 12 above)
Comment (4-life) [inline src/main.mobile.js]: this is for the global scope adguardAssistant

File: test/mobile.html (hunk 13 above)
Comment (4-life) [inline test/mobile.html]: Tests is falling without this call

File: src/button.mobile.js (hunk 11 above)
Comment (4-life) [inline src/button.mobile.js]: but we dont need gmApi in this implementation. Isnt it?

**Pull request merged**